src/heitech.zer0mqXt.core/Main/Zer0Mq.cs

```diff
@@ -0,0 +1,44 @@
+using heitech.zer0mqXt.core.Adapters;
+using heitech.zer0mqXt.core.infrastructure;
+
+namespace heitech.zer0mqXt.core.Main
+{
+    public class Zer0Mq : IZer0MqBuilder
+    {
+        private ILogger _logger;
+        private ISerializerAdapter _serializer;
+        private Zer0Mq()
+        {
+            _logger = new BasicLogger();
+            _serializer = new InternalAdapters();
+        }
+
+        public IZer0MqBuilder SetSerializer(ISerializerAdapter adapter)
+        {
+            _serializer = adapter;
+            return this;
+        }
+
+        public IZer0MqBuilder SetLogger(ILogger adapter)
+        {
+            _logger = adapter;
+            return this;
+        }
+
+        public static IZer0MqBuilder Go() => new Zer0Mq();
```

Comment:
What do you think about an additional overload of this method with `ISerializerAdapter` as parameter.
And the parameterless overload calls that one with `InternalAdapters`?

Reply:
I find it to be more explicit with SetterMethods this way. 

Point taken, that the Go method seems kind of useless here, but again I kinda like how it reads, so I would rather leave it like that.